data/user_data_chef_provision

```diff
@@ -0,0 +1,135 @@
+# get the local hostname from the instance metadata service`, pull out the region and zone letter
+LOCAL_HOSTNAME=`wget -qO- http://instance-data/latest/meta-data/local-hostname`
+
+REGION="`echo \"$AVAILABILITY_ZONE\" | sed -e 's:\([0-9][0-9]*\)[a-z]*\$:\\1:'`"
+
+AVAILABILITY_ZONE_LETTER="`echo $AVAILABILITY_ZONE | sed 's/.*\(.\)/\1/'`"
+
+#Sleep a bit wait for tags to get set.
+sleep 5
+
+# Get the environment from the tag
+ENVIRONMENT=`aws ec2 describe-tags --region $REGION --filter "Name=resource-id,Values=$INSTANCE_ID" "Name=key,Values=Environment" --output=text | sed -r 's/TAGS\t(.*)\t.*\t.*\t(.*)/\2/'`
+# Get the role from the tag
+ROLE=`aws ec2 describe-tags --region $REGION --filter "Name=resource-id,Values=$INSTANCE_ID" "Name=key,Values=Role" --output=text | sed -r 's/TAGS\t(.*)\t.*\t.*\t(.*)/\2/'`
+# Get the group from the tag
+GROUP=`aws ec2 describe-tags --region $REGION --filter "Name=resource-id,Values=$INSTANCE_ID" "Name=key,Values=Group" --output=text | sed -r 's/TAGS\t(.*)\t.*\t.*\t(.*)/\2/'`
+# Pull that stupid role prefix out of the role.
+ROLE_MINUS_ROLE="$(echo $ROLE | sed -r 's/[Role]+//g')"
+# Get the uniqueness from the instance id
+UNIQUENESS="$(echo $INSTANCE_ID | sed -r 's/[i-]+//g')"
+# pull the first letter out of the environment
+ENVIRONMENT_PREFIX="$(echo $ENVIRONMENT | head -c 1)"
+# join them all together into a server name
+SERVER_NAME_MIXED_CASE="$ENVIRONMENT_PREFIX-$GROUP-$ROLE_MINUS_ROLE-use1$AVAILABILITY_ZONE_LETTER-$UNIQUENESS"
+
+# Lowercase that shiz
+SERVER_NAME="$(echo $SERVER_NAME_MIXED_CASE | sed 's/./\L&/g')"
+# Create the name tag
+aws ec2 create-tags --resources $INSTANCE_ID --tags "Key=Name,Value=$SERVER_NAME" --region $REGION
+
+sed -i '/requiretty/d' /etc/sudoers
+
+# set the hostname of the server
+
+case $ENVIRONMENT in
+"prod")
+  SUBDOMAIN=".app.hudl.com"
+  HOSTED_ZONE="Z1LKTAOOYM3H8T"
+  ;;
+"stage")
+  SUBDOMAIN=".app.staghudl.com"
+  HOSTED_ZONE="Z3ETV7KVCRERYL"
+  ;;
+*)
+  SUBDOMAIN=".thorhudl.com"
+  HOSTED_ZONE="ZAH3O4H1900GY"
+  ;;
+esac
+
+FULL_HOSTNAME="$SERVER_NAME$SUBDOMAIN"
+
+mkdir /etc/chef
+
+CHANGE="{\"Comment\": \"Add recordset for $SERVER_NAME\", \"Changes\": [ { \"Action\": \"UPSERT\", \"ResourceRecordSet\": { \"Name\": \"$FULL_HOSTNAME\", \"Type\": \"CNAME\", \"TTL\": 60, \"ResourceRecords\": [ { \"Value\": \"$LOCAL_HOSTNAME\" } ] } } ] }"
+echo $CHANGE > /etc/chef/route53change.json
+
+aws route53 change-resource-record-sets --hosted-zone-id $HOSTED_ZONE --change-batch file:///etc/chef/route53change.json
+
+hostname $FULL_HOSTNAME
+
+sed -i 's/^releasever=latest/# releasever=latest/' /etc/yum.conf
+
+yum clean all
+
+touch /etc/chef/client.rb
+
+# set the ec2 ohai hints so ohai knows it's in a ec2, VPCs mess it up.
+mkdir -p /etc/chef/ohai/hints
+touch /etc/chef/ohai/hints/ec2.json
+
+# Write first boot json
+echo "{\"run_list\": [\"role[$ROLE]\"]}" > /etc/chef/firstrun.json
+
+# Write Validation key to server
+echo '-----BEGIN RSA PRIVATE KEY-----
```

Comment:
@jamiegs I've removed his key pair from the GitHub history, but it may have already been cached elsewhere, so I've also created a new key pair for him. The one that was in the history shouldn't work anymore, even if it exists elsewhere.